ledger_device_sdk/src/nbgl.rs

```diff
@@ -279,8 +200,36 @@ impl<'a> NbglHomeAndSettings<'a> {
         Reply: From<<T as TryFrom<ApduHeader>>::Error>,
     {
         unsafe {
-            if NVM_REF.is_some() {
-                self.settings_contents = nbgl_content_t {
+            loop {
+                let info_contents: Vec<*const c_char> = self
+                    .info_contents
+                    .iter()
+                    .map(|s| s.as_ptr())
+                    .collect::<Vec<_>>();
+
+                let info_list: nbgl_contentInfoList_t = nbgl_contentInfoList_t {
+                    infoTypes: INFO_FIELDS.as_ptr() as *const *const c_char,
+                    infoContents: info_contents[1..].as_ptr() as *const *const c_char,
+                    nbInfos: INFO_FIELDS.len() as u8,
+                };
+
+                let icon: nbgl_icon_details_t = match self.glyph {
+                    Some(g) => g.into(),
+                    None => nbgl_icon_details_t::default(),
```

Comment:
It would be better I think to use NULL pointer if there is no icon to display... this is what the C api expects, not a structure with default values

Reply:
Why feel the need to rewrite my code like I didn't properly think about what I did ?

Reply:
This is managed internally by Rust bindings:
```
impl Default for nbgl_icon_details_s {
    fn default() -> Self {
        let mut s = ::core::mem::MaybeUninit::<Self>::uninit();
        unsafe {
            ::core::ptr::write_bytes(s.as_mut_ptr(), 0, 1);
            s.assume_init()
        }
    }
}
```